### PR TITLE
Validate check name is unique

### PR DESCRIPTION
**Pull request by ryanslade**
Validate that check name is unique before starting the migration.

### CODE REVIEW

File: pkg/state/state.go [hunk 1]
```diff
@@ -542,13 +542,13 @@ func (s *State) Start(ctx context.Context, schemaname string, migration *migrati
 		return nil, err
 	}
 
-	var schema schema.Schema
-	err = json.Unmarshal([]byte(rawSchema), &schema)
+	var unmarshalledSchema schema.Schema
```
Comment (exekias) [inline pkg/state/state.go:545]: this change doesn't seem related to the pr?

**Review by exekias — APPROVED** [review]

File: pkg/state/state.go (hunk 1 above)
Comment (ryanslade) [reply pkg/state/state.go:545]: Just a small rename to stop a warning about shadowing a package name. I felt it was too small to include in a separate PR.
